### PR TITLE
Dashboard: Fix WordPress Admin Menu Scroll. Remove Squish for Dashboard Header.

### PR DESCRIPTION
**Pull request by carloskelly13**
## Summary

Fix container styles to ensure WordPress menu can scroll. Remove scroll hijacking on Dashboard to allow for a natural page flow.

## To-do
Need to remove all the squish code from the layout provider and remove `Layout.Squishable` cus we shouldn't need it any more

## User-facing changes

The WordPress Menu now can scroll if the browser height is short
The Dashboard Header is in-place and the page has a full natural scroll.

## Testing Instructions

1. Launch the Dashboard
2. Resize your browser window vertically so it is shorter
3. Scroll the page and verify the WordPress Admin Menu on the left can be scrolled

1. Launch the Dashboard
2. Scroll the My Stories view down to see the scroll to top button in the bottom right
3. Click that button and verify scrolling back to top works

Verify that there are no regressions when viewing:
- Telemetry Banner
- Toasts
- Any page (My Stories, Templates & Editor Settings)

#5647 
#5874

### CODE REVIEW

**Review by samwhale — COMMENTED** [review]
Scrolling works as you said 🥳 

It looks like the telemetry banner at the top of the page is getting squished:
image

**Comment by maxyinger** [comment]
Yea I'm getting some funkiness too. two major things I see are the divider between dashboard sidenav and dashboard content disappearing when the wp nav is collapsed:
Screen Shot 2021-01-14 at 10 30 19 AM
 
 
 Also seems like the dashboard sidnav isn't supposed to scroll like this:
 Screen Shot 2021-01-14 at 10 29 30 AM

**Comment by swissspidy** [comment]
@littlemilkstudio ^ that's for you

**Comment by maxyinger** [comment]
> @littlemilkstudio ^ that's for you

Right on. thanks for lettin me know. still a gotta go through and remove a bunch of dead code, but will consent when I'm done making changes

**Comment by maxyinger** [comment]
@googlebot I consent.

File: assets/src/dashboard/app/views/editorSettings/components.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ import { Link } from '../../../components/link';
 import { BUTTON_TYPES, KEYBOARD_USER_SELECTOR } from '../../../constants';
 
 export const Wrapper = styled.div`
-  margin: 0 107px;
+  margin: 0;
 `;
```
Comment (swissspidy) [inline assets/src/dashboard/app/views/editorSettings/components.js]: can just do 

```js
export const Wrapper = styled.div``;
```

now, no?

File: assets/src/dashboard/components/scrollToTop/index.js [hunk 2]
```diff
@@ -75,19 +75,23 @@ const DropUpArrowIcon = styled(ChevronLeft)`
...
-  useEffect(() => {
-    const isVisibleFromProgress = (event) => {
-      setIsVisible(event.data.progress > 0);
-    };
+  const handleScroll = useCallback(() => {
+    if (window.scrollY > 0) {
+      setIsVisible(true);
+      return;
+    }
+    setIsVisible(false);
```
Comment (swissspidy) [inline assets/src/dashboard/components/scrollToTop/index.js]:
```suggestion
    setIsVisible(window.scrollY > 0);
```

File: assets/src/dashboard/components/scrollToTop/index.js [hunk 3]
```diff
@@ -75,19 +75,23 @@ const DropUpArrowIcon = styled(ChevronLeft)`
...
+  }, []);
 
-    addSquishListener(isVisibleFromProgress);
-    return () => {
-      removeSquishListener(isVisibleFromProgress);
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
```
Comment (swissspidy) [inline assets/src/dashboard/components/scrollToTop/index.js]:
```suggestion
    window.addEventListener('scroll', handleScroll, { passive: true });
```

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#improving_scrolling_performance_with_passive_listeners

File: assets/src/dashboard/components/scrollToTop/index.js [hunk 4]
```diff
@@ -75,19 +75,23 @@ const DropUpArrowIcon = styled(ChevronLeft)`
...
   const {
-    actions: { scrollToTop, addSquishListener, removeSquishListener },
+    actions: { scrollToTop },
   } = useLayoutContext();
 
-  useEffect(() => {
-    const isVisibleFromProgress = (event) => {
-      setIsVisible(event.data.progress > 0);
-    };
+  const handleScroll = useCallback(() => {
```
Comment (swissspidy) [inline assets/src/dashboard/components/scrollToTop/index.js]: Scroll events can be fired a lot. Might need to consider using `useDebouncedCallback` here.

Comment (maxyinger) [reply assets/src/dashboard/components/scrollToTop/index.js]: yup. good call

File: assets/src/dashboard/app/views/editorSettings/components.js (hunk 1 above)
Comment (maxyinger) [reply assets/src/dashboard/app/views/editorSettings/components.js]: probably. I'll try it out

File: assets/src/dashboard/components/layout/scrollable.js [hunk 5]
```diff
@@ -25,51 +25,18 @@ import styled from 'styled-components';
...
-  bottom: 0;
-  left: 0;
-  overflow-x: hidden;
-  overflow-y: scroll;
-
   /**
    * Adds inertial scrolling to iOS
    * devices like iPad
    */
   -webkit-overflow-scrolling: touch;
```
Comment (BrittanyIRL) [inline assets/src/dashboard/components/layout/scrollable.js]: is this still necessary?

File: assets/src/dashboard/components/pageStructure/index.js [hunk 6]
```diff
@@ -29,6 +29,7 @@ import { __, sprintf } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
+import { ADMIN_TOOLBAR_HEIGHT } from '../../../edit-story/constants';
 import { BEZIER } from '../../../animation';
```
Comment (BrittanyIRL) [inline assets/src/dashboard/components/pageStructure/index.js]: Can you pull this from the design system instead? `THEME_CONSTANTS.WP_ADMIN.TOOLBAR_HEIGHT`

**Review by BrittanyIRL — COMMENTED** [review]
This looks great in the app but i'm getting an error when i try to spin up storybook to make sure nothing is breaking there. Is storybook working for you on this?

File: assets/src/dashboard/components/layout/scrollable.js (hunk 5 above)
Comment (maxyinger) [reply assets/src/dashboard/components/layout/scrollable.js]: nope. good call

**Comment by maxyinger** [comment]
> This looks great in the app but i'm getting an error when i try to spin up storybook to make sure nothing is breaking there. Is storybook working for you on this?

Ope forgot to run the check on storybook. will make sure everything's working there

File: assets/src/dashboard/components/pageStructure/index.js [hunk 7]
```diff
@@ -29,7 +29,7 @@ import { __, sprintf } from '@wordpress/i18n';
 /**
  * Internal dependencies
  */
-import { ADMIN_TOOLBAR_HEIGHT } from '../../../edit-story/constants';
+import { THEME_CONSTANTS } from '../../../design-system';
```
Comment (BrittanyIRL) [inline assets/src/dashboard/components/pageStructure/index.js:32]: 👍  thank you!

**Review by BrittanyIRL — APPROVED** [review]
LGTM!

**Review by embarks — APPROVED** [review]

**Review by zachhale — COMMENTED** [review]
This works great except the dashboard left nav does not scroll vertically and is still cut off. This does not appear to be a new issue from the current behavior, though.

image

**Pull request merged**